src/test/java/namor/template/stack/ArrayStackTest.java

```diff
@@ -103,4 +103,34 @@ public void resizeStackWhenLastCellOfArrayIsFilled() {
         assertThat(actual).isGreaterThan(firstSize);
     }
 
+    @Test
+    public void stackWorksWithString() {
+        ArrayStack<String> stringArrayStack = new ArrayStack<>(3);
+        final String expected = "B";
+        String actual = "D";
+
+        stringArrayStack.push("A");
+        stringArrayStack.push("B");
+        stringArrayStack.push("C");
+        stringArrayStack.pop();
+
+        if(!stringArrayStack.isEmpty()) {
+            actual = stringArrayStack.pop();
+        }
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void stackWorksWithBoolean() {
+        ArrayStack<Boolean> booleanArrayStack = new ArrayStack<>(3);
+        final boolean expected = true;
+        final boolean actual;
+
+        if(booleanArrayStack.isEmpty()) {
+            booleanArrayStack.push(false);
+            booleanArrayStack.push(true);
+        }
+        actual = booleanArrayStack.pop();
+        assertThat(actual).isEqualTo(expected);
+    }
```

Comment:
and here are extra tests. Will repeat again. No need to check if java works fine check only logic you code